### PR TITLE
[RLlib] Deprecate all Model(v1) usage.

### PR DESCRIPTION
**Pull request by sven1977**
This PR removes all Model (v1) dependencies in RLlib and replaces it with the ModelV2 class/API.
A warning is generated whenever the old Model class is used stating that it will be deprecated in the near future.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: rllib/agents/ddpg/OBSOLETED_ddpg_policy.py [hunk 1]
```diff
@@ -1,403 +0,0 @@
-from gym.spaces import Box
```
Comment (sven1977) [inline rllib/agents/ddpg/OBSOLETED_ddpg_policy.py:1]: This seemed to have slipped in somehow a few day ago :(

File: rllib/models/modelv2.py [hunk 2]
```diff
@@ -267,3 +280,98 @@ def __enter__(self):
 
     def __exit__(self, *args):
         pass
+
```
Comment (sven1977) [inline rllib/models/modelv2.py:283]: These helper funds were simply moved here from `model.py`.

File: rllib/models/tf/tf_modelv2.py [hunk 3]
```diff
@@ -52,39 +52,6 @@ def context(self):
         else:
             return ModelV2.context(self)
 
-    def forward(self, input_dict, state, seq_lens):
```
Comment (sven1977) [inline rllib/models/tf/tf_modelv2.py:55]: Duplicate code in a child class.

File: rllib/models/torch/torch_modelv2.py [hunk 4]
```diff
@@ -41,39 +41,6 @@ def __init__(self, *args, **kwargs):
             name,
             framework="torch")
 
-    def forward(self, input_dict, state, seq_lens):
```
Comment (sven1977) [inline rllib/models/torch/torch_modelv2.py:44]: Duplicate code in a child class.

Comment (ericl) [reply rllib/models/torch/torch_modelv2.py:44]: I had this before for clarity. Do you think it makes it less clear how to implement the model if it's removed?

**Review by ericl — APPROVED** [review]
Looks good, just a question about whether we should remove the forward definitions

File: rllib/models/torch/torch_modelv2.py (hunk 4 above)
Comment (sven1977) [reply rllib/models/torch/torch_modelv2.py:44]: Fair point. I'll move it back. Removing the entire method in TorchModel breaks the linter anyways due to some abstract meta-class lint error (I think. it has to do with the fact that TorchModel is an nn.Module as well).

Comment (ericl) [reply rllib/models/torch/torch_modelv2.py:44]: Sounds good. Should also do it for the TF model too then.

**Pull request merged**